pandas/tests/internals/test_external_block.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=W0102
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.internals import Block, BlockManager
+
+
+class CustomBlock(Block):
+
+    def formatting_values(self):
+        return np.array(["Val: {}".format(i) for i in self.values])
+
+
+def test_custom_repr():
+    values = np.arange(3, dtype='int64')
+
+    # series
+    block = CustomBlock(values, placement=slice(0, 3))
```

Comment:
If you don't use fastpath, it does not preserve the Block type. Eg:

```
In [53]: block = CustomBlock(values, placement=slice(0, 3))

In [54]: s = pd.Series(block, index=pd.RangeIndex(3), fastpath=True)

In [55]: s._data._block
Out[55]: CustomBlock: 3 dtype: int64

In [56]: s = pd.Series(block, index=pd.RangeIndex(3))

In [57]: s._data._block
Out[57]: ObjectBlock: 3 dtype: object
```

The reason is that we don't check for `isinstance(data, Block)` in the Series `__init__.py` (we do for `isinstance(data, SingleBlockManager)` in the non-fastpath code path. 

For that reason I am also using the fastpath in GeoPandas.